### ISSUE TITLE
Ingest fails when features are not 100% the same for both adata objects.

### ISSUE THREAD

**Issue by Xparx** [scverse/scanpy#2001]
- [X ] I have checked that this issue has not already been reported.
- [X ] I have confirmed this bug exists on the latest version of scanpy.
- [X ] (optional) I have confirmed this bug exists on the master branch of scanpy.

I'm working on comit `63b42e4b` (latest master).
I'm not sure if intended or not but it seems like it would be usefull if one were able to ingest data that don't share 100% of all features.

### Minimal code sample (that we can copy&paste without having any data)

```python
adata = sc.datasets.paul15()
sc.pp.pca(adata_ref)
sc.pp.neighbors(adata_ref)
sc.tl.leiden(adata_ref)
adata = adata_ref[:, :1000].copy() # assume adata_ref has more than 1000 genes.
sc.tl.ingest(adata, adata_ref, obs='leiden')
```

Error message
```pytb
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-37-b3cd11e67810> in <module>
----> 1 sc.tl.ingest(adata, adata_ref, obs='leiden')

~/projects/scanpy/scanpy/tools/_ingest.py in ingest(adata, adata_ref, obs, embedding_method, labeling_method, neighbors_key, inplace, **kwargs)
    125 
    126     ing = Ingest(adata_ref, neighbors_key)
--> 127     ing.fit(adata)
    128 
    129     for method in embedding_method:

~/projects/scanpy/scanpy/tools/_ingest.py in fit(self, adata_new)
    437 
    438         if not ref_var_names.equals(new_var_names):
--> 439             raise ValueError(
    440                 'Variables in the new adata are different '
    441                 'from variables in the reference adata'

ValueError: Variables in the new adata are different from variables in the reference adata
```

--- 

#### Versions

<details>

sc.logging.print_header()
scanpy==1.8.0.dev78+gc488909a anndata==0.7.6 umap==0.5.0 numpy==1.19.4 scipy==1.5.4 pandas==1.1.4 scikit-learn==0.23.2 statsmodels==0.12.1 python-igraph==0.8.3 leidenalg==0.8.3 pynndescent==0.5.1

</details>

**Comment by potulabe** [comment]
I agree that such an opportunity totally should be, but the example given is not relevant, because to ingest data you have to have all the genes, which were involved in the determination of neighbors, to be present in the query dataset.
The relevant example would be when you somehow managed to count common embeddings (e.g. Harmony-adjusted PCA on the reference's highly variable genes and corresponding Symphony-adjusted PCA on the query), you can then use these embeddings to ingest Umap without the necessity for all the gene_names to be the same.
So maybe the check of equality of gene vars is required only if inside tl.ingest initial embeddings for the query are calculated.
